test/e2e/framework/gateways.go

```diff
@@ -198,3 +198,33 @@ func (f *Framework) GatewayCleanup() {
 		}
 	}
 }
+
+// Perform a gateway failover.
+// The failover for the real environment will crash the gateway node.
+// THe failover for the kind environment will set the "false" label on the gw node.
+func (f *Framework) DoFailover(cluster ClusterIndex, gwNode, gwPod string) {
+	provider := DetectProvider(cluster, gwNode)
+
+	if provider == "kind" {
+		f.SaveGatewayNode(cluster, gwNode)
+		f.SetGatewayLabelOnNode(cluster, gwNode, false)
+	} else {
+		cmd := []string{"sh", "-c", "echo 1 > /proc/sys/kernel/sysrq && echo b > /proc/sysrq-trigger"}
+
+		_, _, err := f.ExecWithOptions(&ExecOptions{
+			Command:       cmd,
+			Namespace:     TestContext.SubmarinerNamespace,
+			PodName:       gwPod,
+			ContainerName: SubmarinerGateway,
+			CaptureStdout: false,
+			CaptureStderr: true,
+		}, cluster)
+		if err != nil {
+			if strings.Contains(err.Error(), "unable to upgrade connection: container not found") {
```

Comment:
Not sure if this message is correct. Is this a typo?

Reply:
@sridhargaddam 
This message is correct.

I'm looking for a specific error message to identify it node crash succeeded.
Any other error message means that crash did not succeeded.

Reply:
Okay, but what does "unable to upgrade connection" mean in this context? The purpose of this API is to trigger failover. So something like "unable to trigger failover" might be more apt.

Reply:
@sridhargaddam 
What happens here, is that I'm sending the crash command to the pod which runs on the required node by using "ExecWithOptions" function.

The function sends the command and expect to receive ack from the pod.
But the node already crashed.
So, it tries to restore the connection.

This error message I get as response from the api.
It's not something I control.

Reply:
Oh ok. So you are comparing the error message. I felt that you were returning the error. Got it.